### PR TITLE
Dampening Sounds

### PR DESCRIPTION
**Pull request by km00700**
Sounds will now be dampened if the player enters rolling stock with
"closed_stock": "true" in the json. Hearable sound level can be set with
"sound_dampening_percentage"

### CODE REVIEW

**Comment by cam72cam** [comment]
This could be simplified by not using a bool to track the dampenSound attribute.

Have it default to 1 (no dampen) and override that when dampening needs to happen.

Also, I don't see a place where it's reset to not dampen in ClientProxy

**Comment by km00700** [comment]
Oops, left the bool. That was from an earlier implementation

File: src/main/java/cam72cam/immersiverailroading/registry/EntityRollingStockDefinition.java [hunk 1]
```diff
@@ -128,6 +130,18 @@ public void parseJson(JsonObject data) throws Exception  {
...
 		
+		dampeningAmount = 0.5f;
+		if (data.has("sound_dampening_percentage")) {
+			dampeningAmount = data.get("sound_dampening_percentage").getAsFloat();
+		}
+		
+		closedStock = true;
+		
+		if (data.has("closed_stock")) {
+			closedStock = data.get("closed_stock").getAsBoolean();
```
[the remark this answers is not in the document]
Comment (km00700) [reply src/main/java/cam72cam/immersiverailroading/registry/EntityRollingStockDefinition.java]: no

File: src/main/java/cam72cam/immersiverailroading/proxy/ClientProxy.java [hunk 2]
```diff
@@ -145,11 +145,27 @@
...
 	private static String missingResources;
+	private static float dampeningAmount = 0.5f;
+	
+	public static float getDampeningAmount() {
+		return dampeningAmount;
+	}
+	
+	public static boolean dampenSound() {
+		EntityPlayerSP player = Minecraft.getMinecraft().player;
+		World world = null;
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/proxy/ClientProxy.java]: Is this variable used?

File: src/main/java/cam72cam/immersiverailroading/sound/ClientSound.java [hunk 3]
```diff
@@ -27,6 +28,7 @@
 	private Vec3d velocity;
 	private float currentPitch = 1;
 	private float currentVolume = 1;
+	private float dampeningFactor = 0.6f;
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/sound/ClientSound.java]: Is this variable used?

**Comment by cam72cam** [comment]
Looks pretty close to ready for merging, just a few small comments.

I'll spin this up tomorrow morning and give it a go

File: src/main/java/cam72cam/immersiverailroading/proxy/ClientProxy.java [hunk 4]
```diff
@@ -145,11 +145,26 @@
 	private static Map<KeyTypes, KeyBinding> keys = new HashMap<KeyTypes, KeyBinding>();
 
 	private static IRSoundManager manager;
-
+	
 	private static MagicEntity magical;
 	public static RenderCacheTimeLimiter renderCacheLimiter = new RenderCacheTimeLimiter();
 
 	private static String missingResources;
+	private static float dampeningAmount = 0.5f;
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/proxy/ClientProxy.java]: default to 1

File: src/main/java/cam72cam/immersiverailroading/proxy/ClientProxy.java [hunk 5]
```diff
@@ -145,11 +145,26 @@
...
+	}
+	
+	public static boolean dampenSound() {
+		EntityPlayerSP player = Minecraft.getMinecraft().player;
+		dampeningAmount = 1.0f;
+		if (player != null && player.isRiding() && player.getRidingEntity() instanceof EntityRidableRollingStock) {
+			EntityRidableRollingStock ridableStock = (EntityRidableRollingStock) player.getRidingEntity();
+			dampeningAmount = ridableStock.getDefinition().dampeningAmount;
+		}
+		return false;
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/proxy/ClientProxy.java]: Don't need to have this return a boolean

File: src/main/java/cam72cam/immersiverailroading/registry/EntityRollingStockDefinition.java [hunk 6]
```diff
@@ -128,6 +129,11 @@ public void parseJson(JsonObject data) throws Exception  {
 		bogeyFront = (float) (data.get("trucks").getAsJsonObject().get("front").getAsFloat() * internal_model_scale);
 		bogeyRear = (float) (data.get("trucks").getAsJsonObject().get("rear").getAsFloat() * internal_model_scale);
 		
+		dampeningAmount = 0.5f;
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/registry/EntityRollingStockDefinition.java]: I played around with it a bit, 0.6 or 0.75 feels like a better default

Comment (km00700) [reply src/main/java/cam72cam/immersiverailroading/registry/EntityRollingStockDefinition.java]: 0.75 is now set as default

**Pull request merged**